omod-2.0/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource2_0.java

```diff
@@ -1,17 +1,197 @@
 import org.openmrs.module.emrapi.db.DbSessionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
 import org.openmrs.module.webservices.rest.web.RestConstants;
 import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.GenericRestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.ArrayProperty;
+import io.swagger.models.properties.BooleanProperty;
+import io.swagger.models.properties.DateProperty;
+import io.swagger.models.properties.RefProperty;
+import io.swagger.models.properties.StringProperty;
 
+/**
+ * {@link Resource} for Attachment, supporting standard CRUD operations
+ */
 @Resource(name = RestConstants.VERSION_1 + "/attachment", supportedClass = Attachment.class, supportedOpenmrsVersions = {
         "2.0.0" })
 public class AttachmentResource2_0 extends AttachmentResource1_10 {
 	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getRepresentationDescription(org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("dateTime");
+			description.addProperty("comment");
+			description.addProperty("complexData");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("dateTime");
+			description.addProperty("comment");
+			description.addProperty("auditInfo");
+			description.addProperty("complexData");
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("comment");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
```

Comment:
I would return the `DefaultRepresentation` incase the current instance of `rep` is unknown; @mks-d what do you think of that?

Reply:
@samuelmale  on which line are your referring to

Reply:
>    return null;

The above. We would have something like:

```java

if (rep instanceof RefRepresentation) {
   // Handle RefRepresentation
   // .......
} else if (rep instanceof FullRepresentation) {
   // Handle FullRepresentation
   // .......
} else {
   // Handle Default
   // .......
}

```
However I don't know what @mks-d thinks about this?

Reply:
Thanks @samuelmale  lets here from @mks-d suggestion

Reply:
Hello @mks-d  @samuelmale @reagan-meant  any suggestion about this ticket, to me i think its ready to be merged.